src/DashboardService.Data/TaskRepository.cs

```diff
@@ -18,28 +21,114 @@ public TaskRepository(IDataProvider provider)
     _provider = provider;
   }
   
-  public Task<Guid?> CreateAsync(DbTask dbDepartment)
+  public async Task<Guid> CreateAsync(DbTask dbTask)
   {
-    throw new NotImplementedException();
+    _provider.Tasks.Add(dbTask);
+    await _provider.SaveAsync();
+
+    return dbTask.Id;
   }
 
-  public Task<List<DbTask>> GetAllAsync(GetTasksFilter filter)
+  public async Task<(List<DbTask> dbTasks, int totalCount)> GetAllAsync(GetTasksFilter filter, CancellationToken ct)
   {
-    throw new NotImplementedException();
+    IQueryable<DbTask> dbTasks = _provider.Tasks.Where(t => !filter.GroupId.HasValue || t.GroupId == filter.GroupId).AsNoTracking();
+
+    if (filter.BoardId.HasValue)
+    {
+      DbBoard dbBoard = await _provider.Boards
+        .Include(b => b.Groups)
+        .FirstOrDefaultAsync(b => b.Id == filter.BoardId, ct);
+      
+      dbTasks = dbTasks
+        .Include(t => t.Group)
```

Comment:
you can add ThenInclude() method to include board of group and simply use .Where(t => t.Board.Group.Id == filter.BoardId.Value)

Reply:
if i understood correctly, we dont need to use then include because BoardId of Group already has a right value. Changed filtering. Am i right and will it work? Or if we will not use thenInclude then BoardId will have incorrect value?

Reply:
yep, missed the fact group just has board id)